Source/VMware.vSphereDSC/DSCResources/VMHostSyslog.ps1

```diff
@@ -0,0 +1,203 @@
+#>
+
+[DscResource()]
+class VMHostSyslog : VMHostBaseDSC {
+    <#
+    .DESCRIPTION
+
+    The remote host(s) to send logs to.
+    #>
+    [DscProperty()]
+    [string] $Loghost
+
+    <#
+    .DESCRIPTION
+
+    Verify remote SSL certificates against the local CA Store.
+    #>
+    [DscProperty()]
+    [bool] $CheckSslCerts = $true
+
+    <#
+    .DESCRIPTION
+
+    Default network retry timeout in seconds if a remote server fails to respond.
+    #>
+    [DscProperty()]
+    [long] $DefaultTimeout = 180
+
+    <#
+    .DESCRIPTION
+
+    Message queue capacity after which messages are dropped.
+    #>
+    [DscProperty()]
+    [long] $QueueDropMark = 90
+
+    <#
+    .DESCRIPTION
+
+    The directory to output local logs to.
+    #>
+    [DscProperty()]
+    [string] $Logdir = '/scratch/log'
+
+    <#
+    .DESCRIPTION
+
+    Place logs in a unique subdirectory of logdir, based on hostname.
+    #>
+    [DscProperty()]
+    [bool] $LogdirUnique = $false
+
+    <#
+    .DESCRIPTION
+
+    Default number of rotated local logs to keep.
+    #>
+    [DscProperty()]
+    [long] $DefaultRotate = 8
+
+    <#
+    .DESCRIPTION
+
+    Default size of local logs before rotation, in KiB.
+    #>
+    [DscProperty()]
+    [long] $DefaultSize = 1024
+
+    <#
+    .DESCRIPTION
+
+    Number of rotated dropped log files to keep.
+    #>
+    [DscProperty()]
+    [long] $DropLogRotate = 10
+
+    <#
+    .DESCRIPTION
+
+    Size of dropped log file before rotation, in KiB.
+    #>
+    [DscProperty()]
+    [long] $DropLogSize = 100
+
+    [void] Set() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        $this.UpdateVMHostSyslog($vmHost)
+    }
+
+    [bool] Test() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        return !$this.ShouldUpdateVMHostSyslog($vmHost)
+    }
+
+    [VMHostSyslog] Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $syslog = [VMHostSyslog]::new()
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.PopulateResult($vmHost, $syslog)
+
+        return $syslog
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value if VMHostSyslog needs to be updated.
```

Comment:
Change to: Returns a boolean value indicating if VMHostSyslog needs to be updated.